### DISCUSSION TITLE
Public data with an owner?

### DISCUSSION THREAD

**Discussion by ondrej-langr** [evoluhq/evolu#180] [Q&A]
Hey,
Love the direction this is rolling. I was tinkering with this for some time and I was wondering how would you set public data with an owner? I only found out that all data are encrypted and not publicly sharable to other users (technically users can sync with your server, but thats about it - atleast to my knowledge)

Typical scenarion would be platform like Twitter. User writes a "post" and wants engagement on their "post" - aka likes, comments, etc... Another scenario could be messaging app.

Is this something that its possible ATM? If not then would it be even supported? (you kind of can achieve this with gun.js)

Thanks 🫶

**Comment by steida** [comment]
I think sharing data with other people and sharing data publicly are two different use cases than owning data, and therefore different means are required. For sharing among a few members, there is public/private cryptography. For public data, there are endless other options. Evolu definitely has plans, which will be shared lately.

**Reply by ondrej-langr** [reply]
Shared (allowing just specific subset of users) and public data are definitelly different things, but kind of necessary today. My question was kind of leaning towards both of these. 

Even though my knowledge is limited in this kind of stuff I would definitelly would like to push some of these narratives further. I will be keeping eye on those plans you are mentioning and we will see what can I help with. 

For now I work on some touchups for server package in my spare time - just adding some fetch handler (basically Next.js route.ts support), express middleware and other so its more versatile and not tied just to express. Will be posting PR soon so we can talk about it there (I hope you are not working on it already though 😁)

**Reply by steida** [reply]
@ondrej-langr I'm working on React Native support. Then explain everything in docs, then new features :-)
